### ISSUE TITLE
Piracy

### ISSUE THREAD

**Issue by JeffreyStrahm** [siggame/MegaMinerAI-Game-Ideas#46]
# **The game:**
## **TLDR**
- Gain infamy to win, max 500 turns.
- 40x40 map with water and islands
- merchant NPCs
- 3 units, port, ship, and crew.
- Gain gold to purchase units
- bury gold or kill merchants/opponent to gain gold they carried

## **Original story abridged: (read if you wish)**
Piracy has a simple concept, you were a member of the West India Trading Company, but as you were offloading disaster struck! The East Indian Trading Company bought you out and every other competitor, leaving you jobless. They even blacklisted you! With no money, you can't get cargo to sell. There is a simple solution, however, you just need to use good old honest piracy! You and the remains of your company take to the seas to start raiding! Sadly a giant kracken sinks your ship, killing your glorious captain. You make it back to shore with the rest of your loot and look to pick a new captain, you obviously. However, not everyone thinks so. Unable to come to an agreement, and after some drunken fighting, you both slip into two rival groups. You both swear eternal war on each other.
link to the original story: [https://github.com/siggame/Megaminer-7/blob/master/client/TheStorySoFar.pdf](url)

## **New Story:**
_I wanted to re-write the story because while that story was fun, it was very long and doesn't fit on a flier._
Your and another crewmate served under the great pirate king of the Indies, but sadly his reign came abruptly to an end due to drinking too much rum and falling off the boat. After his death, no one can decide on who the new pirate king, so you both separate off, swearing you will show them that you are the greater pirate captain and that you shall be the next pirate king!

## **Victory:**
The game continues till either one player doesn't have any ships and can not get enough gold to their base to build one, or when 500 turns have passed.
If 500 turns have passed, the player with the most infamy win.
Otherwise, it is whoever collected the most gold in total.
Otherwise, it is whoever has the most value of units, including ports.
Otherwise, it is whoever has the most health in total.
Otherwise, it is a coin flip.

## **Map:**
The map is a random generation of land masses with sea separating them. This map is 40 x 40. There should be 5 merchant ports. Each port will automatically trade with it's two closest neighbors by sending ships at a pre-determined interval.

## **Land:**
At every point of land adjacent(N, S, E, W) to the sea, you can build a port. Ports should cost about 100 GP. Definitely more expensive than pirates. Ports allow you to move gold on and off your ship. Without a port, each pirate can bring 10 gold with him. All units also heal twice as fast on a port, and a port must be destroyed before a ship can be attacked. Pirates otherwise have no carry limit and all extra gold will be left on the shore where the pirate stood. At any point on land, you may choose to bury your gold, placing an X on the map. Gold will grow exponentially based on distance from the nearest player unit. NPCs don't count. You can bury and dig up gold as free actions.

Each player will also start with a home port on opposite ends, your crew cannot take damage on this port. The dock it'self cannot be destroyed. You must bring gold here to spend it to build ships, crew, and ports.

## **Merchants:**
Merchants control ports and will send out ships to their neighbors. The longer they trade without losing a ship, the more the ships are worth. The more frequently they are attacked, the more guards there are on the ships. Like treasure piles but they will stab you if you try and loot them. These ships should carry a large amount of gold and a few units. They should also grow in difficulty, and loot, as the game goes on. Merchants will also be aggressive to whoever is in the lead infamy wise.

## **Units:**
In the game, you can have ships and crew. Ships can attack ships and crew, but at reduced effectiveness. However, crew can only attack other crew. Crew can only be on land or on a boat, and a boat can only be in a friendly port or at sea. (when ports are built, they are treated as sea tile when it comes to building other ports). Crew can bury and pick up gold, and will also take gold off of defeated pirates. There is no carry limit for this. Crew can exist on the same tile, and their stats are added together when they stand together. They can be split, but are otherwise treated as one unit. When they are split hp is shared as evenly as possible, favoring the original. Splitting takes a move action from both unit, you merge with the target, and you cannot merge if you have unequal actions left. AKA you must have the same number of moves left and you both must have either attacked or both not attacked. Crew die in combat if there wouldn't be enough HP for each crew to have 1.
Each crew should cost 20GP
Each ship should cost 1000GP
Ships have no Crew limit.
Units will heal 1 hit point per turn they have been out of combat. A ship is in combat for 2 turns after attacking a unit. If crew in combat board a ship out of combat, it is brought it.

| Name | HP | Damage | movement | Range | cost |
|----|----|----|----|----|----|
|**Ship**|10|2/1|4|3|1000GP|
|**Crew**|4|1|2|1|20GP|
|**Port**|10|N/A|N/A|N/A|100GP|

## **Combat + infamy:**
No pirate has every gain infamy from hoarding gold, no they gain infamy for ruling to the high Cs! You gain infamy based on what units you kill, ships should be worth a lot more infamies than crew. This is then scaled based on whether or not you outnumbered the enemy, less infamy, or if they outnumbered you, more infamy. This is not calculated by total population but by units within the combat. Note, if your rival attack a merchant ship and flees and you immediately dive in and kill it, he will still gain infamy!

Note: When calculating infamy, the final additions don't get calculated until the participants are out of combat. Chaining combats can lead to interesting results. Crew on a ship in combat are part of that combat.

When it comes to fighting, you just add up the damage from one side and subtract if from the HP of the other. Should you kill them, you get all the gold they carried plus a bounty worth 1/10th their value. Units can move after attacking. Note, Pirates can attack from one ship to the other, and if you land crew on a ship that has none, you gain immediate ownership. You do not gain infamy from it, however.

Crew can attack ports and other crew, ships can attack everything.

**The bad news:**
There is no slogan or logo for this game. So here we go:
Time to sail on the high Cs!
logo:
![test](https://user-images.githubusercontent.com/22081179/35548680-cc497fb2-0546-11e8-8f87-0ce030456b15.png)
(Warning, this is a concept, someone who is artsy should really redo it)

**Comment by JeffreyStrahm** [comment]
Time to sail the 3 C's!

**Comment by JeffreyStrahm** [comment]
![skull_wrench](https://user-images.githubusercontent.com/22081179/36506846-66ac3d50-171d-11e8-9a3d-26aa230f4e3d.jpg)
Here it is, I had little time and little skill, but here!

**Comment by ShimmeringHydra** [comment]
- ARRRGS
- Sail the high Cs
- If we fall off the edge of the world, it's your segfault
- Recurse the high Cs
- Let the bounty overflow
- You wouldn't download a ship
- something something something something something something something something something something torrenting

**Comment by aws9t5** [comment]
AAARRRRGGGSSSS you ready kids.
SYSTEM CRASH on the PORT bow.

**Comment by JenniferColeman** [comment]
![piratebird](https://user-images.githubusercontent.com/34194946/36749839-93711718-1bc1-11e8-84de-443778cfb80f.png)
Here's a pirate bird I drew I think looks kinda cute!

**Comment by LanChauLeTran** [comment]
Slogan Idea: 
Rated Arrgh
